### DISCUSSION TITLE
CTkToplevel out of focus

### DISCUSSION THREAD

**Discussion by KotikLina** [TomSchimansky/CustomTkinter#1671] [Q&A]
When a new window is created, it is not in focus. I had to use some "crutches" to keep it in focus when created. I want to know if this can be fixed without using these "crutches". I don't speak English, so I singled out the word "crutches" because of the heterogeneity of its meanings. The code has redundant elements,
on which the program is based, but I would like to replace them with something that could represent normal code
```py
def OpenGameWindow(self):
    if self.toplevel_gameWindow is None or not self.toplevel_gameWindow.winfo_exists():
        self.toplevel_gameWindow = PlayWindow(focus=True)  # create window if its None or destroyed
    else:
        self.toplevel_gameWindow.focus()  # if window exists focus it


class PlayWindow(CTkToplevel):
    def __init__(self, focus=False):
        super().__init__()
        if focus:
            self.after(250, self.focus_force)  # Call focus_force() after 250ms
```
Also I would like to fix the miss click when clicking on the "exit" button. I fixed it with this code:
```py
        self.ButtonStartGame3 = CTkButton(master=self.FrameMainMenu, text="Выход", font=self.font2,
                                          command=self.CloseWindow)
        self.ButtonStartGame3.grid(row=3, column=0, ipadx=20, ipady=2, padx=20, pady=0)

    def CloseWindow(self):
        self.after(100, self.destroy)
```

**Comment by avalon60** [comment]
Is there any reason that you can't do a `self.grab_set()`
?

**Reply by KotikLina** [reply]
That's not going to work for me.

**Comment by Akascape** [comment]
@KotikLina 
Idk, but this issue is showing after the 5.1.3 update.
But we can fix this using:
```python
toplevel.transient(root)
toplevel.resizable(True, True)
```
Or
```python
toplevel.attributes('-topmost', 1)
toplevel.attributes('-topmost', 0)
```

**Reply by KotikLina** [reply]
I'm afraid that's not it. I either lose control over the other windows or my window is on top of all the other windows, but I don't need that
